From 5de558383472895c0375279c4d0984cacd907f2d Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 31 Jan 2020 15:06:12 +0200
Subject: [PATCH] [SW-8228] arcboot: zero btlver in scratchpad

Boot flow is without BTL so just pass zero version

TODO: Add in main loop ability to provide arcboot version upon request

Change-Id: I5386a97673bc08c7c34b41c74fbb16fd2517f6bf
---

diff --git a/Makefile b/Makefile
index 58b8d1a..9b3562f 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@
 export CC AR RANLIB
 OBJCOPY = $(CROSS_COMPILE)objcopy
 
-ARCBOOT_VER := 0x0$(shell cat localversion-habanalabs)
+ARCBOOT_VER := 0x0
 
 LIBDIR = sdk/lib
 ARCBOOT_CFLAGS = -Wall -W -Werror -O3
diff --git a/app/arcboot_ram.c b/app/arcboot_ram.c
index b9f2300..507ac6b 100644
--- a/app/arcboot_ram.c
+++ b/app/arcboot_ram.c
@@ -71,9 +71,9 @@
 	unsigned int arcboot_id = BTLID;
 
 	/*
-	 * Using scratchpad registers for
-	 * passing the current running BTL ID to CPU SW
-	 * ------------------------------
+	 * Traditionally we used scratchpad registers for
+	 * passing the current running BTL ID and VER to CPU SW
+	 * TODO: Need do it in case of ARCBOOT
 	 */
 	regmap_writel(arcboot_id, mmPSOC_GLOBAL_CONF_SCRATCHPAD_31);
 
